README.md

```diff
@@ -1 +1,100 @@
+
+**image_1** and **image_2** can be either 3D or 4D images.  If they are 3D, then topup is estimated, and by default these two are corrected.  If they are 4D, then the first volume from each acquisition is used for the topup calculation.  Then, topup correction is applied to the full 4D image, unless **topup_only** is selected from the config file.
+
+
+#### Optional
+* **apply_to_1**: A third image to apply the topup correction calculated from **image_1** and **image_2**.  This image must have the same PE direction as **image_1**
+* **apply_to_2**: A fourth image to apply the topup correction calculated from **image_1** and **image_2**.  This image must have the same PE direction as **image_2**
+* **config_file**: a config file to pass into topup to use with the [--config](https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/topup/TopupUsersGuide#Configuration_files) option.
+
+### Config settings
+* **gear-log-level**: Gear Log verbosity level (ERROR|WARNING|INFO|DEBUG)
+* **topup_only**: only run topup and get fieldmaps (do no correct images, either 4D inputs, or additional **apply_to_X** images)
+* **displacement_field** save displacement fields (hidden topup option *--dfout*)
+* **jacobian_determinants** save jacobian determinants (hidden topup option *--jacout*)
+* **rigid_body_matrix** save rigid body transformation matricies to align volumes (hidden topup option *--rbmout*)
+* **verbose** output verbose information to the log (topup option *--verbose*)
+* **topup_debug_level** Topup Log verbosity level (0|1|2|3) (hidden topup option *--debug*).  **WARNING** this produces a LOT of additional files.
+* **QA** Save a topup QA image comparing distorted to corrected images
+
+
+
+### Outputs
+
+Depending on the settings in the gear's conguration, you will have some of the following files:
+
+
+#### *Constant Output*
+* **topup_log.txt**: topup's log file
+* **topup_fmap.nii.gz**: output-file containing the off-resonance field.
+* **topup_fieldcoef.nii.gz**: output-files containing the spline coefficients for the off-resonance field
+* **topup_movpar.txt**: The transformation parameters to fit **image_2** to **image_1**
+* **topup_input_corrected.nii.gz**: The corrected version of the merged **image_1** and **image_2** used for topup's *--imain* input option
+
+
+#### *Output with **apply_to_X** options, and **topup_only** is false*
+* **topup_corrected_<apply_to_1>.nii.gz**: Topup corrected image from **apply_to_1**
+* **topup_corrected_<apply_to_2>.nii.gz**: Topup corrected image from **apply_to_2**  
+*note that if **image_1** and **image_2** are 4D, there will be additional files like these, which are the corrected inputs.  The basename for each input image is appended to the end of 'topup_corrected'*
```

Comment:
Yeah I'm not explaining this the best.  You're SUPPOSED to use two single-volume SBRef images to calculate the correction field.  You're supposed to calculate the field from these two images, and then use that to correct for your 4D fMRI timeseries.  If you don't have those, but you have two 4D-fMRI time series' with opposite phase encoding directions, you can just pull one volume from each scan and use them like the single volume SBRef images.  They won't have as good contrast, but it can work.  So this code assumes that if your "image 1" and "Image 2" scans are 4D, you'll want to both calculate the distortion field AND correct all the volumes in the scans.

Reply:
But the output are 4D right (not split 3D volumes right)?

Reply:
There are two kinds of outputs: 1) Corrected versions of the images in "input1", "input2", "applyto1", and "applyto2".   These will match the dimension of the associated input.  2) optionally, you can also have the gear save several 3D volumes, which would be the fieldmap, or warp field, or other (depending on config settings), that was used to correct the other input images.

Reply:
I've changed this to:
note that if **image-1** and **image-2** are 4D, this gear will also create TOPUP corrected versions of **image-1** and **image-2**.  The output name will be topup-corrected-<image-1>.